src/HIE/Bios/Cradle.hs

```diff
@@ -466,13 +470,53 @@ cabalProcess workDir command args = do
   let newEnvironment = extraEnvironmentArgs ++ environment
   wrapper_fp <- withCabalWrapperTool ("ghc", []) workDir
   buildDir <- cabalBuildDir workDir
-  let cabalArgs = ["--builddir=" <> buildDir, command, "--with-compiler", wrapper_fp] ++ args
-  let cabalProc = proc "cabal" cabalArgs
+  let baseCabalArgs = ["--builddir=" <> buildDir, command, "--with-compiler", wrapper_fp]
+  extraCabalArgs <- case ghcDirLoadResult of
+        CradleSuccess (ghcBin, libdir) -> do
+          ghcPkgPath <- withGhcPkgTool ghcBin libdir
+          pure ["--with-hc-pkg", ghcPkgPath]
+        _ -> pure []
+  let cabalProc = proc "cabal" (baseCabalArgs ++ extraCabalArgs ++ args)
   pure $ CradleSuccess (cabalProc
       { env = Just newEnvironment
       , cwd = Just workDir
       })
 
+withGhcPkgTool :: FilePath -> FilePath -> IO FilePath
+withGhcPkgTool ghcPathAbs libdir = do
+  let ghcName = takeFileName ghcPathAbs
+  -- TODO: check for existence
+  let ghcPkgPath = guessGhcPkgFromGhc ghcName
+  if isWindows
+    then pure ghcPkgPath
+    else withWrapperTool ghcPkgPath
+  where
+    ghcDir = takeDirectory ghcPathAbs
+
+    guessGhcPkgFromGhc ghcName =
+      case parseCompilerId (T.pack ghcName) of
```

Comment:
Maybe just strip the `ghc` prefix and prepend `ghc-pkg`? Then we can omit the parsing I guess.

Reply:
What should happen in the 'Nothing' case?

Reply:
`fail` hard

Reply:
or wait... it's possible that distros use arch triple prefix, e.g. `x86_64-redhat-linux-ghc`... even ghcup supports prefixed cross GHCs, so the most robust way would be to just `s/ghc/ghc-pkg/`... not assuming prefix.

Reply:
On the other hand:

```
echo 'exeprog="ghc-stage1"'                          >> "/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/bin/arm-linux-gnueabihf-ghc-8.10.7"
echo 'executablename="$exedir/$exeprog"'           >> "/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/bin/arm-linux-gnueabihf-ghc-8.10.7"
echo 'datadir="/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/share"'                             >> "/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/bin/arm-linux-gnueabihf-ghc-8.10.7"
echo 'bindir="/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/bin"'                               >> "/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/bin/arm-linux-gnueabihf-ghc-8.10.7"
echo 'topdir="/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/lib/arm-linux-gnueabihf-ghc-8.10.7"'                               >> "/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/bin/arm-linux-gnueabihf-ghc-8.10.7"
echo 'executablename="$exedir/ghc"' >> "/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/bin/arm-linux-gnueabihf-ghc-8.10.7"
cat ghc/ghc.wrapper                         >> "/builds/haskell/ghcup-hs/.ghcup/ghc/arm-linux-gnueabihf-8.10.7/bin/arm-linux-gnueabihf-ghc-8.10.7"
```

I don't have the actual tarball at hand, but this indicates that for a cross ghc, the actual binary name is `ghc-stage1`. This supports my theory that we shouldn't parse the version. Not assuming a prefix might make it additionally more robust.

Reply:
Replace sounds like a good idea.

Reply:
`replace` sounds fine. Changed it, although what happens if "ghc" doesn't occur in the name? Should we check for that case as well?

Reply:
> `replace` sounds fine. Changed it, although what happens if "ghc" doesn't occur in the name? Should we check for that case as well?

we could just let the world descend into chaos